### ISSUE TITLE
About DFDC 

### ISSUE THREAD

**Issue by IItaly** [polimi-ispl/icpr2020dfdc#24]
Hey @zhangconghhh ,

> In your ICPR paper, you mention the test result on the dfdc dataset is 0.8782. The test dataset in your paper is the test_sample_videos part or the last 10 veidos in the train_sample_videos? Because I saw there is a separation in the split.py.

> ![image](https://user-images.githubusercontent.com/23146227/80863629-4223da00-8cb0-11ea-9b48-ecf97a0cd7ae.png)

as you can find in the paper, we used only the videos from the training set for training, validation and testing. In particular, we used the videos from the first 35 folders as training set, videos from folder 35 to 40 as validation set, and finally videos from the last 10 folders as test set. 
At the time the paper has been written, the challenge wasn't closed already, so we didn't have at hand the videos from the `test_sample_videos` folder you cited previously in your comments.

> And when I run the make_dataset.sh for the DFDC dataset, I also meet a problem.
image
I use the path to train_sample_videos as the DFDC path. And I got the json_path as 'PosixPath('/media/disk/Backup/zhangcong/deepfake/dfdc/train_sample_videos/metadata.json')'.
I can't get a int type for the df_tmp['folder'].
Can you help me with that?

I'm sorry, I think I didn't understand your question. In `make_dataset.sh` from the last release we execute first `index_dfdc.py` and then `extract_faces.py`; in `index_dfdc.py` however, you just need to specify the path to the folder containing the 50 folders of videos of the DFDC training set, _already unzipped_. 
The metadata json file the script elaborates is then taken from each one of the 50 folders and is used to create an overall Pandas Dataframe with info about the whole dataset. You don't need to specify any json path anywhere in the code. I'll recap this for you quickly.
Before launching `make_dataset.sh`:

1. You should have downloaded the DFDC dataset from Kaggle;
2. You should have unzipped all the 50 folders contained in the `train_sample_videos` folder;
3. You should run `index_dfdc.py` indicating as argument the path to the folder containing the 50 folders of the DFDC training set. 

Hope this helps. Have a good weekend!
Cheers

Edoardo

_Originally posted by @CrohnEngineer in https://github.com/polimi-ispl/icpr2020dfdc/issues/2#issuecomment-622972096_

It seems that the DFDC dateset has been diffrent from what you said. I downloaded the DFDC dataset,there're not 50 folders in the train_sample videos. there are just 401 mp4 files.So I guess that it is the puzzle that the questioner haved.

**Comment by IItaly** [comment]
> Hey @zhangconghhh ,
> 
> > In your ICPR paper, you mention the test result on the dfdc dataset is 0.8782. The test dataset in your paper is the test_sample_videos part or the last 10 veidos in the train_sample_videos? Because I saw there is a separation in the split.py.
> 
> > ![image](https://user-images.githubusercontent.com/23146227/80863629-4223da00-8cb0-11ea-9b48-ecf97a0cd7ae.png)
> 
> as you can find in the paper, we used only the videos from the training set for training, validation and testing. In particular, we used the videos from the first 35 folders as training set, videos from folder 35 to 40 as validation set, and finally videos from the last 10 folders as test set.
> At the time the paper has been written, the challenge wasn't closed already, so we didn't have at hand the videos from the `test_sample_videos` folder you cited previously in your comments.
> 
> > And when I run the make_dataset.sh for the DFDC dataset, I also meet a problem.
> > image
> > I use the path to train_sample_videos as the DFDC path. And I got the json_path as 'PosixPath('/media/disk/Backup/zhangcong/deepfake/dfdc/train_sample_videos/metadata.json')'.
> > I can't get a int type for the df_tmp['folder'].
> > Can you help me with that?
> 
> I'm sorry, I think I didn't understand your question. In `make_dataset.sh` from the last release we execute first `index_dfdc.py` and then `extract_faces.py`; in `index_dfdc.py` however, you just need to specify the path to the folder containing the 50 folders of videos of the DFDC training set, _already unzipped_.
> The metadata json file the script elaborates is then taken from each one of the 50 folders and is used to create an overall Pandas Dataframe with info about the whole dataset. You don't need to specify any json path anywhere in the code. I'll recap this for you quickly.
> Before launching `make_dataset.sh`:
> 
> 1. You should have downloaded the DFDC dataset from Kaggle;
> 2. You should have unzipped all the 50 folders contained in the `train_sample_videos` folder;
> 3. You should run `index_dfdc.py` indicating as argument the path to the folder containing the 50 folders of the DFDC training set.
> 
> Hope this helps. Have a good weekend!
> Cheers
> 
> Edoardo
> 
> _Originally posted by @CrohnEngineer in [#2 (comment)](https://github.com/polimi-ispl/icpr2020dfdc/issues/2#issuecomment-622972096)_

It seems that the DFDC dateset has been diffrent from what you said. I downloaded the DFDC dataset,there're not 50 folders in the train_sample videos. there are just 401 mp4 files.So I guess that it is the puzzle that the questioner haved.

**Comment by CrohnEngineer** [comment]
Hey @IItaly ,

> there are just 401 mp4 files

are you sure about the dataset you have?
The DFDC dataset, in its original version used for the challenge, counts roughly **120000** videos (yeah, you read it right, it is that huge :P and you can check it [here](https://arxiv.org/abs/2006.07397)).

> there're not 50 folders in the train_sample videos.

I don't know if the people managing the dataset have changed its structure now, but at the time of writing the paper that was the way it was organized. Looking at the [Kaggle page](https://www.kaggle.com/c/deepfake-detection-challenge/data) where the train dataset is hosted and you can download it, it seems to me that the organization in 50 folders is not changed.
Bests,

Edoardo

**Comment by IItaly** [comment]
> 嘿@IItaly
> 
> > 只有401 MP4文件
> 
> 你确定你拥有的数据集吗？
> DFDC数据集，在其用于挑战的原始版本，计数约**12万**视频（是的，你读对了，它是巨大的:P，你可以[在这里](https://arxiv.org/abs/2006.07397)检查它）。
> 
> > train_sample视频中没有50个文件夹。
> 
> 我不知道管理数据集的人现在是否改变了其结构，但在撰写本文时，它就是这样组织的。查看托管列车数据集的[Kaggle页面](https://www.kaggle.com/c/deepfake-detection-challenge/data)，您可以下载它，在我看来，50个文件夹中的组织没有改变。
> 贝斯特，
> 
> 爱德华多

yeah I mean that maybe someone downloaded the sample dataset so that they will get the error like this:
![image](https://user-images.githubusercontent.com/24908762/112131356-71af3e00-8c04-11eb-8185-b67a3bb719fe.png)
